staging/src/k8s.io/pod-security-admission/policy/visitor.go

```diff
@@ -20,18 +20,19 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-// ContainerVisitor is called with each container and the field.Path to that container
-type ContainerVisitor func(container *corev1.Container)
+// ContainerVisitorWithPath is called with each container and the field.Path to that container.
+type ContainerVisitorWithPath func(container *corev1.Container, pathFn PathFn)
 
-// visitContainers invokes the visitor function for every container in the given pod spec
-func visitContainers(podSpec *corev1.PodSpec, visitor ContainerVisitor) {
+// visitContainersWithPath invokes the visitor function with a pointer to the spec
+// of every container in the given pod spec and the field.Path to that container.
+func visitContainersWithPath(podSpec *corev1.PodSpec, visitor ContainerVisitorWithPath) {
 	for i := range podSpec.InitContainers {
-		visitor(&podSpec.InitContainers[i])
+		visitor(&podSpec.InitContainers[i], initContainersFldPath.index(i))
```

Comment:
Here, if we have the options, we can pass `nil` directly instead of the path function.

Reply:
that's a good idea